src/components/StudyForm/StudyForm.tsx

```diff
@@ -120,7 +125,26 @@ const StudyForm: React.FC<StudyFormProps> = ({ submission, onSave }) => {
               onIonBlur={field.onBlur}
               onIonInput={field.onChange}
               {...field}
-            />
+            >
+              {/* If the user is logged in, show a button they can press to use their name. */}
+              {loggedInUser !== null ? (
+                <IonButton
+                  fill={"clear"}
+                  slot={"end"}
+                  title={"Use my name"}
+                  aria-label={"Use my name"}
+                  onClick={() => field.onChange(loggedInUser.name)}
```

Comment:
I think invoking `onChange` manually is discouraged (https://react-hook-form.com/docs/usecontroller/controller). I would use the `setValue` function which is already being unpacked from the `useForm` call.

```suggestion
                  onClick={() => setValue(field.name, loggedInUser.name)}
```

Reply:
Thanks, @pkalita-lbl. I was confused about that. I interpreted the `Controller` documentation (which I think has a grammatical error in it) as saying:

> "...you should avoid manually invoking `setValue`..."

<img width="930" alt="image" src="https://github.com/user-attachments/assets/bdf1e459-e0c9-4554-a889-a413dfc8e402">

That led to me searching for alternatives. I saw, in other places on the (documentation) page, they kept using `field.onChange`, so that's why I went with that one.

I'll switch to `setValue`. Maybe the author of the documentation meant to write:

```diff
This prop update `formState` and you should avoid 
- manually invoke 
+ manually invoking it and, instead, invoke
`setValue` or other API related to field update.
```

Reply:
Yeah I'm nearly certain that's what the documentation is trying to say. The `react-hook-form` documentation unfortunately has a lot of such typos.